src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,7 +39,15 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+        String[] values = trimmedAddress.split(",");
         this.value = trimmedAddress;
+        if (values.length != 4) {
```

Comment:
take note of the use of magic numbers, you've declared the `fields` number above but its not used.

Reply:
oh haha sorry i forgot